flax/nnx/transforms/iteration.py

```diff
@@ -1350,6 +1374,10 @@ def scan_wrapper(*args, **kwargs):
       assert input_carry_argnum is None
       pure_carry_arg = None
 
+    if len(carry_graphdefs) > 0:
+      # We should store carry graphdefs separately as it can change in the scan call
+      scan_fn.carry_graphdefs = carry_graphdefs
```

Comment:
we can avoid this inplace mutation if we just move the definition of `scan_fn` into `scan_wrapper`.